### PR TITLE
Vt merge

### PR DESCRIPTION
**Pull request by SlovenlyTF**
## Description

Minor bug fixed surrounding: 
estimatedHours value 
Lectures
Components
status 
Exercise
Add som error codes 
 E11 - Component Errors

## Changes

- Minor bug fixed surrounding: 
- estimatedHours value 
- Lectures
- Components
- status 
- Exercise
- Add som error codes
- E11 - Component Errors

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

### CODE REVIEW

**Comment by SlovenlyTF** [comment]
We it's failing... We've been trying to make it work all day, and need to run it past gamefied, to see how we can fix it.

File: __tests__/routes/courseRoutes.test.js [hunk 1]
```diff
@@ -747,14 +747,15 @@ describe('Course Routes', () => {
       const response = await request(app)
         .patch('/api/courses/' + fakeCourse._id)
         .set('Authorization', `Bearer ${token}`)
-        .send({ title: 'Test', category: 'sewing', difficulty: 1, description: 'Sewing test', estimatedHours: 2 })
+        .send({ title: 'Test', category: 'sewing', difficulty: 1, description: 'Sewing test'})
         .expect(200);
 
+        console.log(response);
```
Comment (MortenMunk) [inline __tests__/routes/courseRoutes.test.js:753]: 😢 Please remove console.log

**Review by MortenMunk — CHANGES_REQUESTED** [review]
Some minor changes requested... I did not run the code locally, and some of the code I am not familiar with, but all in all it looks fine, except for the comments 👍 🥇

File: __tests__/routes/courseRoutes.test.js (hunk 1 above)
Comment (magnemo) [reply __tests__/routes/courseRoutes.test.js]: Has been done

**Review by Mafusn — APPROVED** [review]
Looks good

**Review by MortenMunk — APPROVED** [review]
Approving, as long as moving of error messages does not cause problems for other groups 👍 🥇

**Review by PrometheusBork — APPROVED** [review]
Looks fine and works with mobile dev for me, but I have a question. 
What is it that you are doing with "estimatedHours"? are you removing it? @SlovenlyTF

**Pull request merged**